essm/equations/_core.py

```diff
@@ -155,23 +155,35 @@ def __add__(self, other):
         )
 
     def subs(self, *args, **kwargs):  # should mirror sympy.core.basic.subs
-        """Return a new equation with subs applied to both sides.
+        r"""Return a new equation with subs applied to both sides.
 
         **Examples:**
 
-        >>> from essm.equations.physics.thermodynamics import eq_Pa, eq_PN2_PO2
-        >>> from essm.variables.physics.thermodynamics import P_N2, P_O2
-        >>> eq_Pa.subs({P_N2: P_O2})
-        Eq(P_a, 2*P_O2 + P_wa)
-        >>> eq_Pa.subs(eq_PN2_PO2)
-        Eq(P_a, P_O2*x_N2/x_O2 + P_O2 + P_wa)
-
+        >>> from essm.equations.physics.thermodynamics import eq_Le,\
+            eq_Dva, eq_alphaa
+        >>> from essm.variables.physics.thermodynamics import Le, D_va,\
+            alpha_a, T_a
+        >>> eq_Le.subs(D_va, eq_Dva.rhs)
+        Eq(Le, alpha_a/(T_a*p_Dva1 - p_Dva2))
+        >>> eq_Le.subs({D_va: eq_Dva.rhs, alpha_a: eq_alphaa.rhs})
+        Eq(Le, (T_a*p_alpha1 - p_alpha2)/(T_a*p_Dva1 - p_Dva2))
+        >>> eq_Le.subs(eq_Dva, eq_alphaa)
+        Eq(Le, (T_a*p_alpha1 - p_alpha2)/(T_a*p_Dva1 - p_Dva2))
         """
         if len(args) == 1:
             if isinstance(args[0], Eq):
                 arg1 = {args[0].lhs: args[0].rhs}
                 return Eq(self.lhs.subs(arg1, **kwargs),
                           self.rhs.subs(arg1, **kwargs))
+        else:
```

Comment:
Using `else:` with `return` will make line 187 unreachable.

a) use `elif`
b) remove `else`, remove one level of indentation and remove 187-189.

Reply:
Line 187 should be reached if *args is a dictionary. Then, len(args) = 1, but it is not an equation, so the else statement will be omitted.

Reply:
Just to clarify, the original `.subs()` takes either a dictionary or a list of tuples or two items, the first of which will be substituted by the second in the expression. With the new definition of `.subs()` it is possible to provide 1, 2 or many equations instead.

Reply:
Dammit, it still only takes up to two equations. Haven't tested for this. Do you know how to extend it to many equations? I will provide an example.

Reply:
Just checked again, and this works: 
```python
from essm.equations.leaf.energy_water import eq_Rs_enbal, eq_El, eq_Hl, eq_Rll
eq_Rs_enbal.subs(eq_El, eq_Hl, eq_Rll)
```
returns:
> Eq(R_s, E_lmol*M_w*lambda_E + a_sh*epsilon_l*sigm*(T_l**4 - T_w**4) + a_sh*h_c*(-T_a + T_l))

I can't put it into the docstring, as the line is too long and if I add a line breat, the test fails.

Reply:
So probably you want to use `elif`
```suggestion
        elif not kwargs:
```

run your current code with following cases:
```
args = (eq, ), kwargs = {} -> if
args = (), kwargs = {} -> else
args = (), kwargs = {foo: bar} -> else
```